src/libutil/util.cc

```diff
@@ -574,6 +574,14 @@ Path getHome()
     static Path homeDir = []()
     {
         auto homeDir = getEnv("HOME");
+        if (homeDir) {
+            // Only use $HOME if it exists and is owned by the current user.
```

Comment:
To clarify, you're not saying to create `$HOME` if it doesn't exist (this would be up to the consumer, i suppose), just to go ahead and return that value if it's set and either it doesn't exist OR it exists, is reachable, and is owned by the current user, correct?